roles/cfg_openwrt/templates/corerouter/config/dhcp.j2

```diff
@@ -75,3 +75,25 @@ config odhcpd 'odhcpd'
 	option leasefile '/tmp/hosts/odhcpd'
 	option leasetrigger '/usr/sbin/odhcpd-update'
 	option loglevel '4'
+
+{% for dhcp_static_lease in dhcp_static_leases | default([]) %}
+{% if dhcp_static_lease['name'] is defined and dhcp_static_lease['mac'] is defined and dhcp_static_lease['ip'] is defined %}
+config host
+        option name '{{ dhcp_static_lease['name'] }}'
+        option mac '{{ dhcp_static_lease['mac'] }}'
+        option ip '{{ dhcp_static_lease['ip'] }}'
+    {% if dhcp_static_lease['dns'] is defined %}
+        option dns '{{ dhcp_static_lease['dns'] }}'
+    {% endif %}
+
+{% endif %}
+{% endfor %}
+
+{% for dhcp_custom_domain_name in dhcp_custom_domain_names | default([]) %}
+{% if dhcp_custom_domain_name['name'] is defined and dhcp_custom_domain_name['ip'] is defined %}
+config domain
+        option name '{{ dhcp_custom_domain_name['name'] }}'
+        option ip '{{ dhcp_custom_domain_name['ip'] }}'
+
```

Comment:
First, L67 should be ```config domain``` but aside from that, it might be a good idea to check that hostnames are not repeated here from the assignments list.  And if a hostname is repeated, but with a different ip address (making sure to compare ipv4 with ipv4 and ipv6 with ipv6 addresses), then report it as an error.